docs/CONTRIBUTING.md

```diff
@@ -56,18 +56,41 @@ Vector team member will find this document useful.
 
 ### New sources, sinks, and transforms
 
-If you're contributing a new source, sink, or transform to Vector, thank you that's way cool! There's a few steps you
-need to think about if you want to make sure we can merge your contribution. We're here to help you along with these steps,
-but they are a blocker to getting a new integration released.
-
-To merge a new source, sink, or transform, you need to:
+If you're thinking of contributing a new source, sink, or transform to Vector, thank you that's way cool! There's a few
+steps you need to think about if you want to make sure we can merge your contribution. We're here to help you along with
+these steps, but they are a blocker to getting a new integration released.
+
+Prior to beginning work on a new source or sink if a GitHub Issue does not already exist, please open one to discuss the
+introduction of the new integration. Maintainers will review the proposal with the following checklist in mind, try and
+consider them when sharing your proposal to reduce the amount of time it takes to review your proposal. This list is not
+exhaustive, and may be updated over time.
+
+- [ ] Can the proposed component’s functionality be replicated by an existing component, with a specific configuration?
+(ex: Azure Event Hub as a `kafka` sink configuration)
+  - [ ] Alternatively implemented as a wrapper around an existing component. (ex. `axiom` wrapping `elasticsearch`)
+- [ ] Can an existing component replicate the proposed component’s functionality, with non-breaking changes?
+- [ ] Can an existing component be rewritten in a more generic fashion to cover both the existing and proposed functions?
+- [ ] Is the proposed component generically usable or is it specific to a particular service?
+  - [ ] How established is the target of the integration, what is the relative market share of the integrated service?
+- [ ] Is there sufficient demand for the component?
+  - [ ] If the integration can be served with a workaround or more generic component, how painful is this for users?
+- [ ] Is the contribution from an individual or the organization owning the integrated service? (examples of
+organization backed integrations: `databend` sink, `axiom` sink)
+  - [ ] Is the contributor committed to maintaining the integration if it is accepted?
```

Comment:
I like that approach. However, it requires some changes in Vector. We need to maintain responsible persons for each "externally supported" plugin. Right now such information is not available in Vector. Maybe some "owner.md" file for each sink and/or labels like "externally supported by <GitHub username>" would help with that.

Reply:
We do have a few components that are "externally supported" but aren't necessarily marked as such, it's been informal but we could make it more trackable.

Reply:
Agreed, it's been informal to date.